src/connector/src/sink/coordinate.rs

```diff
@@ -77,6 +78,20 @@ impl<W: SinkWriter<CommitMetadata = Option<SinkMetadata>>> LogSinker for Coordin
             .await?;
         let mut sink_writer = self.writer;
         log_reader.start_from(log_store_rewind_start_epoch).await?;
+        let first_item = log_reader.next_item().await?;
+        if let (Some(log_store_rewind_start_epoch), (first_epoch, _)) =
+            (log_store_rewind_start_epoch, &first_item)
+            && log_store_rewind_start_epoch >= *first_epoch
+        {
+            bail!(
+                "log_store_rewind_start_epoch {} not later than first_epoch {}",
+                log_store_rewind_start_epoch,
+                first_epoch
+            );
+        }
```

Comment:
Should we check` log_store_rewind_start_epoch < first_epoch` before calling `start_from`?